eslint.config.js

```diff
@@ -0,0 +1,193 @@
+      'mocha-no-only/mocha-no-only': ['error'],
+      'no-bitwise': 2,
+      camelcase: 2,
+      curly: 2,
+      eqeqeq: 2,
+      'guard-for-in': 2,
+      'wrap-iife': [2, 'any'],
+      'no-use-before-define': [
+        2,
+        {
+          functions: false
+        }
+      ],
+      'new-cap': 2,
+      'no-caller': 2,
+      'no-empty': 2,
+      'no-new': 2,
+      'no-plusplus': 0,
+      'no-undef': 2,
+      'no-unused-vars': 2,
+      strict: 0,
+      'max-params': [2, 6],
+      'max-depth': [2, 5],
+      'max-len': 0,
+      semi: 0,
+      'no-cond-assign': 0,
+      'no-debugger': 2,
+      'no-eq-null': 0,
+      'no-eval': 2,
+      'no-unused-expressions': 0,
+      'block-scoped-var': 0,
+      'no-iterator': 0,
+      'linebreak-style': 0,
+      'no-loop-func': 0,
+      'no-multi-str': 0,
+      'no-proto': 0,
+      'no-script-url': 0,
+      'dot-notation': 2,
+      'no-new-func': 0,
+      'no-new-wrappers': 0,
+      'no-shadow': 2,
+      'no-restricted-syntax': [
+        'error',
+        {
+          selector: 'MemberExpression[property.name=tagName]',
+          message: "Don't use node.tagName, use node.nodeName instead."
+        },
+        // node.attributes can be clobbered so is unsafe to use
+        // @see https://github.com/dequelabs/axe-core/pull/1432
+        {
+          // node.attributes
+          selector:
+            'MemberExpression[object.name=node][property.name=attributes]',
+          message:
+            "Don't use node.attributes, use node.hasAttributes() or axe.utils.getNodeAttributes(node) instead."
+        },
+        {
+          // vNode.actualNode.attributes
+          selector:
+            'MemberExpression[object.property.name=actualNode][property.name=attributes]',
+          message:
+            "Don't use node.attributes, use node.hasAttributes() or axe.utils.getNodeAttributes(node) instead."
+        },
+        // node.contains doesn't work with shadow dom
+        // @see https://github.com/dequelabs/axe-core/issues/4194
+        {
+          // node.contains()
+          selector:
+            'CallExpression[callee.object.name=node][callee.property.name=contains]',
+          message:
+            "Don't use node.contains(node2) as it doesn't work across shadow DOM. Use axe.utils.contains(node, node2) instead."
+        },
+        {
+          // vNode.actualNode.contains()
+          selector:
+            'CallExpression[callee.object.property.name=actualNode][callee.property.name=contains]',
+          message:
+            "Don't use node.contains(node2) as it doesn't work across shadow DOM. Use axe.utils.contains(node, node2) instead."
+        }
+      ]
+    },
+    plugins: {
+      'mocha-no-only': mochaNoOnly
+    }
+  },
+  {
+    languageOptions: {
+      ecmaVersion: 2023,
+      globals: {
+        axe: true,
+        Promise: true,
+        ...globals.node,
+        ...globals.es2015
+      }
+    }
+  },
+  {
+    files: ['lib/**/*.js'],
```

Comment:
This needs [to ignore the reporters and check after methods](https://github.com/dequelabs/axe-core/blob/develop/.eslintrc.js#L99C23-L99C72) as we don't want them to have access to `window` or `document`.

```suggestion
    files: ['lib/**/*.js'],
    // reporters and check after methods should not have access to window or document
    ignores: ['lib/core/reporters/**/*.js', 'lib/**/*-after.js'],
```